graphics.cpp

```diff
@@ -72,3 +72,59 @@ GLubyte *Texture::data() const {
 
 	return texture_data;
 }
+
```

Comment:
I try to avoid operator overloading as much as possible, since it obscures function calls and this makes telling what is _actually_ going on in some code much harder.

A vector type like this is probably one of the most sensible uses but I still dislike it.

Reply:
Oh yeah totally, you can still have convenience functions that hide all of that mess. I'm just saying, I would rather have:
```cpp
void add(Point &l, const Point &r);
Point sum(const Point &l, const Point &r);
```
instead of...
```cpp
Point &operator +=(Point &l, const Point &r);
Point operator +(Point &l, const Point &r);
```
I will admit that with the non-operator approach, the actual operation is a little less obvious because of the naming. But I still prefer it.

Reply:
Hm, one thing I didn't immediately think about is because `Point` is not a real class, any functions we write for it have to be free, like the operators you made here. A free function that's just called `add` is really dumb. A free overload for the + operator is significantly less dumb.

Given that, here's what I'll say. I want to make some changes to the point class, which might come through in another PR, so for now, let's just leave things as you've made them here.